tasks/sanity-check.yml

```diff
@@ -11,13 +11,50 @@
     - "{{ jenkins_jcasc_config_file }}"
     - "{{ jenkins_jcasc_version }}"
     - "{{ jenkins_pimt_version }}"
-    - "{{ jenkins_plugins_file }}"
     - "{{ jenkins_port }}"
     - "{{ jenkins_prefix }}"
     - "{{ jenkins_root }}"
     - "{{ jenkins_user }}"
     - "{{ jenkins_version }}"
 
+- name: Assert that both JCasC configuration file variables aren't conflicting
+  ansible.builtin.fail:
+    msg: "jenkins_jcasc_config_file and jenkins_jcasc_config_files are mutually exclusive"
+  when:
+    - jenkins_jcasc_config_file != ""
+    - jenkins_jcasc_config_files | length > 0
+
+- name: Set jenkins_jcasc_config_files if necessary
+  ansible.builtin.set_fact:
+    jenkins_jcasc_config_files:
+      - "{{ jenkins_jcasc_config_file }}"
+  when: jenkins_jcasc_config_file != ""
+
+- name: Ensure that configuration files are defined
+  ansible.builtin.assert:
+    that: jenkins_jcasc_config_files | length > 0
+    quiet: true
+
+- name: Ensure that configuration files have unique names
+  block:
+    - name: Initialize jenkins_jcasc_config_filenames fact
+      ansible.builtin.set_fact:
+        jenkins_jcasc_config_filenames: []
+
+    - name: Assemble list of configuration filenames
+      ansible.builtin.set_fact:
+        jenkins_jcasc_config_filenames: >
+          {{ jenkins_jcasc_config_filenames + [filename | basename] }}
```

Comment:
We should also do the `| replace('.j2', '')` here, otherwise we will get a bad surprise if both `jenkins.yaml` and `jenkins.yaml.j2` are in `jenkins_jcasc_config_files`.

Reply:
Fixed in 82979b34d0c82f8772754a7986d4f13f41b8801b.